packages/openapi-generator/src/wrapper-files/tsconfig-json.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @experimental This API is experimental and might change in newer versions. Use with caution.
+
+ */
+import { readJSON } from '@sap-cloud-sdk/util';
+import { GeneratorOptions } from '../options';
+
+export function tsconfigJson(options: GeneratorOptions): string {
+  let tsConfig = {
+    compilerOptions: {
+      target: 'es5',
+      module: 'commonjs',
+      outDir: 'dist',
+      lib: ['esnext', 'dom'],
+      declaration: true,
+      declarationMap: true,
+      sourceMap: true,
+      diagnostics: true,
+      moduleResolution: 'node',
+      esModuleInterop: true,
+      inlineSources: false,
+      strict: true
+    },
+    include: ['**/*.ts'],
+    exclude: ['dist/**/*', 'test/**/*', '**/*.spec.ts', 'node_modules/**/*']
+  };
+  if (options.tsConfig) {
+    tsConfig = readJSON(options.tsConfig) as any;
```

Comment:
I first had the merge version. But since both objects can have subobjects a proper merge is not so simple. I think if your are at the step, that you want to adjust the default tsconfig I would say you either take the existing one as a starting point or you are able to understand the compile error and adjust the config accordingly.

Reply:
I think that makes sense until proven otherwise :)